core/src/main/java/org/opensearch/sql/expression/parse/GrokExpression.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.parse;
+
+import io.krakens.grok.api.Grok;
+import io.krakens.grok.api.GrokCompiler;
+import io.krakens.grok.api.Match;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.expression.Expression;
+
+/**
+ * GrokExpression with grok patterns.
+ */
+@EqualsAndHashCode(callSuper = true)
+@ToString
+public class GrokExpression extends ParseExpression {
+  private static final Logger log = LogManager.getLogger(GrokExpression.class);
+  private static final GrokCompiler grokCompiler = GrokCompiler.newInstance();
+
+  static {
+    grokCompiler.registerDefaultPatterns();
+  }
+
+  @EqualsAndHashCode.Exclude
+  private final Grok grok;
+
+  /**
+   * GrokExpression.
+   *
+   * @param sourceField source text field
+   * @param pattern     pattern used for parsing
+   * @param identifier  derived field
+   */
+  public GrokExpression(Expression sourceField, Expression pattern, Expression identifier) {
+    super("grok", sourceField, pattern, identifier);
+    this.grok = grokCompiler.compile(pattern.valueOf(null).stringValue());
+  }
+
+  @Override
+  ExprValue parseValue(ExprValue value) {
+    String rawString = value.stringValue();
+    Match grokMatch = grok.match(rawString);
+    Map<String, Object> capture = grokMatch.capture();
+    Object match = capture.get(identifierStr);
+    if (match != null) {
+      return new ExprStringValue(match.toString());
+    }
+    log.warn("failed to extract pattern {} from input {}", grok.getOriginalGrokPattern(),
```

Comment:
we should sanitize rawString when logging

Reply:
is it warn? or debug?

Reply:
changed to debug. rawString is unstructured text not sure how to sanitize, let me know if i should remove it from log

Reply:
we should replace rawString as `***`.

Reply:
hard coded it to `***`